contentcuration/contentcuration/tests/viewsets/test_recommendations.py

```diff
@@ -125,3 +129,247 @@ def test_recommendation_generic_error(self, mock_load_recommendations):
+        )
+        self.assertEqual(response.status_code, 201, response.content)
+
+    def test_list_fails(self):
+        response = self.client.get(reverse("recommendations-interaction-list"), format="json")
+        self.assertEqual(response.status_code, 405, response.content)
+
+    def test_retrieve_fails(self):
+        recommendations_interaction = RecommendationsInteractionEvent.objects.create(
+            **{
+                'context': {'test_key': 'test_value'},
+                'contentnode_id': self.interaction_node.id,
+                'content_id': self.interaction_node.content_id,
+                'feedback_type': 'IGNORED',
+                'feedback_reason': '----',
+                'recommendation_event_id': self.recommendation_event.id
+            }
+        )
+        response = self.client.get(reverse("recommendations-interaction-detail", kwargs={"pk": recommendations_interaction.id}), format="json")
+        self.assertEqual(response.status_code, 405, response.content)
+
+    def test_update_recommendations_interaction(self):
+        recommendations_interaction = RecommendationsInteractionEvent.objects.create(
+            **{
+                'context': {'test_key': 'test_value'},
+                'contentnode_id': self.interaction_node.id,
+                'content_id': self.interaction_node.content_id,
+                'feedback_type': 'IGNORED',
+                'feedback_reason': '----',
+                'recommendation_event_id': self.recommendation_event.id
+            }
+        )
+        updated_data = self.recommendations_interaction_object
+        updated_data['feedback_type'] = 'PREVIEWED'
+        response = self.client.put(
+            reverse("recommendations-interaction-detail", kwargs={"pk": recommendations_interaction.id}),
+            updated_data,
+            format="json"
+        )
+        self.assertEqual(response.status_code, 200, response.content)
+
+    def test_partial_update_recommendations_interaction(self):
+        recommendations_interaction = RecommendationsInteractionEvent.objects.create(
+            **{
+                'context': {'test_key': 'test_value'},
+                'contentnode_id': self.interaction_node.id,
+                'content_id': self.interaction_node.content_id,
+                'feedback_type': 'IGNORED',
+                'feedback_reason': '----',
+                'recommendation_event_id': self.recommendation_event.id
+            }
+        )
+        response = self.client.patch(
+            reverse("recommendations-interaction-detail", kwargs={"pk": recommendations_interaction.id}),
+            {'feedback_type': 'IMPORTED'},
+            format="json"
+        )
+        self.assertEqual(response.status_code, 200, response.content)
+
+    def test_destroy_recommendations_interaction(self):
+        recommendations_interaction = RecommendationsInteractionEvent.objects.create(
+            **{
+                'context': {'test_key': 'test_value'},
```

Comment:
Not worth changing, but food for thought: use the kwargs directly instead of destructuring a dict. That way you eliminate needing to wrap the keys in quotes.